src/nodes/display/PixelationNode.js

```diff
@@ -0,0 +1,239 @@
+
+		this._createEdgesRT = new RenderTarget();
+		this._createEdgesRT.texture.name = 'PixelationNode.renderEdges';
+		this._lowerResolutionRT = new RenderTarget();
+		this._lowerResolutionRT.texture.name = 'PixelationNode.lowerResolution';
+
+		// Output textures
+
+		this._outputTextureNode = passTexture( this, this._lowerResolutionRT.texture );
+
+		this.updateBeforeType = NodeUpdateType.RENDER;
+
+	}
+
+	setSize( width, height ) {
+
+		this._createEdgesRT.setSize( width, height );
+		this._lowerResolutionRT.setSize( width, height );
+
+	}
+
+	updateBefore( frame ) {
+
+		const { renderer } = frame;
+
+		const textureNode = this.textureNode;
+		const map = textureNode.value;
+
+		// Set resolution uniform
+
+		const adjustedWidth = map.image.width / this.pixelSizeNode.value;
+		const adjustedHeight = map.image.height / this.pixelSizeNode.value;
+		this._resolution.value.set( adjustedWidth, adjustedHeight, 1 / adjustedWidth, 1 / adjustedHeight );
+
+		const currentRenderTarget = renderer.getRenderTarget();
+		const currentMRT = renderer.getMRT();
+		const currentTexture = textureNode.value;
+
+		createEdgesQuad.material = this._createEdgesMaterial;
+		lowerResolutionQuad.material = this._lowerResolutionMaterial;
+
+		// Set size of render edges to match size of initial render target.
+
+		this.setSize( map.image.width, map.image.height );
+
+		const textureType = map.type;
+		this._createEdgesRT.texture.type = textureType;
+		this._lowerResolutionRT.texture.type = textureType;
+
+		// Apply create edges post-process step to createEdgesQuad.
+
+		renderer.setMRT( null );
+		renderer.setRenderTarget( this._createEdgesRT );
+		createEdgesQuad.render( renderer );
+
+		// Set input of next pass to output of last step.
+		textureNode.value = this._createEdgesRT.texture;
+
+		// Apply lower resolution post-process step to lowerResolutionQuad.
+
+		renderer.setRenderTarget( this._lowerResolutionRT );
+		lowerResolutionQuad.render( renderer );
+
+		// Reset render target and MRT back to initial values.
+
+		renderer.setRenderTarget( currentRenderTarget );
+		renderer.setMRT( currentMRT );
+
+		// Set textureNode back to intial input texture for next pass.
+
+		textureNode.value = currentTexture;
+
+	}
+
+	setup( builder ) {
+
+		const { textureNode, depthNode, normalNode } = this;
+
+		const uvNodeTexture = textureNode.uvNode || uv();
+		const uvNodeDepth = depthNode.uvNode || uv();
+		const uvNodeNormal = normalNode.uvNode || uv();
+
+		const pixelizeUV = ( coord ) => floor( coord.mul( this._resolution.xy ) ).div( this._resolution.xy );
+
+		const sampleTexture = () => textureNode.uv( uvNodeTexture );
+
+		const samplePixel = () => textureNode.uv( pixelizeUV( uvNodeTexture ) );
+
+		const sampleDepth = ( x, y ) => depthNode.uv( uvNodeDepth.add( vec2( x, y ).mul( this._resolution.zw ) ) ).r;
+
+		const sampleNormal = ( x, y ) => normalNode.uv( uvNodeNormal.add( vec2( x, y ).mul( this._resolution.zw ) ) ).rgb.mul( 2.0 ).oneMinus();
```

Comment:
If you write the line like so, it should be possible to remove the `directionToColor()` call.
```js
const sampleNormal = ( x, y ) => normalNode.uv( uvNodeNormal.add( vec2( x, y ).mul( this._resolution.zw ) ) ).rgb.normalize();
```
At least on my system I get better results which edges closer to the original.